### PR TITLE
update performance hint and number of infer request to OV API 2.0 in CPU plugin

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *refactor performance hint and number of infer requests in CPU plugin to OV API 2.0*
 - *update check for number of streams and performance hint*

### Tickets:
 - *CVS-112654*

### CODE REVIEW

File: src/plugins/intel_cpu/src/config.cpp [hunk 1]
```diff
@@ -70,15 +70,37 @@ void Config::applyDebugCapsProperties() {
...
             streamExecutorConfig.SetConfig(key, val);
-        } else if (hintsConfigKeys.end() != std::find(hintsConfigKeys.begin(), hintsConfigKeys.end(), key)) {
-            perfHintsConfig.SetConfig(key, val);
+        } else if (key == ov::hint::performance_mode.name()) {
+            const auto prefHint = ov::util::from_string(val, ov::hint::performance_mode);
+            OPENVINO_SUPPRESS_DEPRECATED_START
+            if (prefHint == ov::hint::PerformanceMode::LATENCY || prefHint == ov::hint::PerformanceMode::THROUGHPUT) {
+                performanceHint = prefHint;
+                changedPerformanceHint = true;
+            } else if (prefHint != ov::hint::PerformanceMode::UNDEFINED) {
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/plugins/intel_cpu/src/config.cpp]: OV API planned to convert `ov::hint::PerformanceMode::UNDEFINED` to `ov::hint::PerformanceMode::LATENCY`. 

Update logic in CPU plugin

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 2]
```diff
@@ -273,17 +274,22 @@ void Engine::ApplyPerformanceHints(std::map<std::string, std::string> &config, c
...
     // save hints parameters to model rt_info
     ov::AnyMap hints_props;
     std::string hint_name;
     const int latency_streams = get_num_numa_nodes();
     int streams;
+
     if (config.streamExecutorConfig._streams_changed) {
         streams = config.streamExecutorConfig._streams;
-    } else if (perf_hint_name == CONFIG_VALUE(LATENCY)) {
+        if (streams <= getAvailableNUMANodes().size()) {
```
Comment (sunxiaoxia2022) [inline src/plugins/intel_cpu/src/plugin.cpp:285]: getAvailableNUMANodes() is old interface, now it is latency_streams.

Comment (wangleis) [reply src/plugins/intel_cpu/src/plugin.cpp]: updated

**Review by sunxiaoxia2022 — APPROVED** [review]

File: src/plugins/intel_cpu/src/config.cpp [hunk 3]
```diff
@@ -74,16 +74,41 @@ void Config::applyDebugCapsProperties() {
...
+            const auto prefHint = ov::util::from_string(val, ov::hint::performance_mode);
+            OPENVINO_SUPPRESS_DEPRECATED_START
+            if (prefHint == ov::hint::PerformanceMode::LATENCY || prefHint == ov::hint::PerformanceMode::UNDEFINED) {
+                performanceHint = ov::hint::PerformanceMode::LATENCY;
+                changedPerformanceHint = true;
+            } else if (prefHint == ov::hint::PerformanceMode::THROUGHPUT) {
+                performanceHint = ov::hint::PerformanceMode::THROUGHPUT;
+                changedPerformanceHint = true;
+            } else {
+                IE_THROW() << "Wrong value " << val << "for property key " << ov::hint::performance_mode.name()
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/config.cpp:94]: Could we use new interface `OPENVINO_THROW` to replace `IE_THROW`?

**Review by riverlijunjie — APPROVED** [review]

**Comment by wenjiew** [comment]
Is this needed for 2023.1?

**Pull request closed without merging**